asteroid/data/avspeech_dataset.py

```diff
@@ -0,0 +1,238 @@
+        return buffer_video
+
+    def get_audio(self):
+        return self.audio
+
+    def get_spec(self):
+            return np.load(self.spec_path)
+
+    def is_spec(self):
+        return self._is_spec
+
+    @staticmethod
+    def load_audio(audio_path: str, sr=16_000, load_spec=False):
+        audio_path = Path(audio_path)
+        spec_exists = False
+        spec_path = Path(*audio_path.parts[:-2], "spec", audio_path.stem + ".npy")
+        if load_spec and spec_path.is_file():
+            spec_exists = True
+            audio = np.load(spec_path)
+        else:
+            audio = librosa.load(str(audio_path), sr=sr)[0]
+        return audio, spec_exists, spec_path
+
+
+class AVSpeechDataset(data.Dataset):
+
+    def __init__(self, input_df_path: Path,
+                 embed_dir: Path,
+                 input_audio_size=2):
+        """
+
+            Args:
+                input_df_path: path for combination dataset
+                input_audio_size: total audio/video inputs
+        """
+        self.input_audio_size = input_audio_size
+        self.embed_dir = embed_dir
+        self.input_df = pd.read_csv(input_df_path.as_posix())
+        self.stft_encoder = Encoder(STFTFB(n_filters=512, kernel_size=400,
+                                           stride=160))
+
+    @staticmethod
+    def encode(x: np.ndarray, p=0.3, stft_encoder=None):
+        if stft_encoder is None:
+            stft_encoder = Encoder(STFTFB(n_filters=512, kernel_size=400,
+                                          stride=160))
+
+        x = torch.from_numpy(x).float()
+
+        # time domain to time-frequency representation
+        tf_rep = stft_encoder(x).squeeze(0) + EPS
+        # power law on complex numbers
+        tf_rep = (torch.abs(tf_rep) ** p) * torch.sign(tf_rep)
+        # (514, 298) -> (257, 298, 2)
+        tf_rep = transforms.to_torchaudio(tf_rep)
+        return tf_rep
+
+    @staticmethod
+    def decode(tf_rep: np.ndarray, p=0.3, stft_decoder=None, final_len=48000):
+        if stft_decoder is None:
+            stft_decoder = Decoder(STFTFB(n_filters=512, kernel_size=400,
+                                          stride=160))
+
+        tf_rep = torch.from_numpy(tf_rep).float()
+
+        # (257, 298, 2) -> (514, 298)
+        tf_rep = transforms.from_torchaudio(tf_rep) + EPS
+        # power law on complex numbers
+        tf_rep = (torch.abs(tf_rep) ** (1/p)) * torch.sign(tf_rep)
+        # time domain to time-frequency representation
+        x = stft_decoder(tf_rep)
+
+        length = len(x)
+        if length != final_len:
+            x = F.pad(x, [0, final_len-length])
+        return x
+
+    def __len__(self):
+        return len(self.input_df)
+
+    def __getitem__(self, idx):
+        row = self.input_df.iloc[idx, :]
+        all_signals = []
+
+        for i in range(self.input_audio_size):
+            #get audio, video path from combination dataframe
```

Comment:
The combination dataset has `video 1, video 2, audio 1, audio 2` columns or something like this?
It would be clearer to have names instead of these indices IMO.